### DISCUSSION TITLE
Announcing TIDAL SDK for iOS and TIDAL SDK for Android

### DISCUSSION THREAD

**Discussion by stefanmoro** [tidal-music/discussions#70] [Announcements]
Following the public release of the [TIDAL SDK for Web](https://developer.tidal.com/blog/tidal-sdk-for-web), we are now pleased to announce expanded platform support by releasing corresponding implementations for [Android](https://github.com/tidal-music/tidal-sdk-android) and [iOS](https://github.com/tidal-music/tidal-sdk-ios). These mobile SDKs unlock similar features for mobile, as we already have for Web. Check out [the announcement](https://developer.tidal.com/blog/tidal-sdk-for-mobile) for more details and how to get started.

Throughout 2024 we will open up more of our platform. We are currently hard at work focusing on several of the requested API features in this forum. Make sure to drop in every once in a while to stay updated!

**Comment by OlofT** [comment]
Looking through the docs one can see that Authorization Code Flow (user login) is "Only available for TIDAL internally developed applications for now". 

Does this not make the whole SDK pointless? The user cannot see their playlist, songs nor play them... Am I missing something?

Question 2: If not, is this part of the "more" you will open up during 2024?

**Reply by ForMiSoft** [reply]
No, only device login is for their internal use. You can get your playlists. But for now you can't change them via api.

**Reply by OlofT** [reply]
No, I can't get the playlist since the users can't login/identify. 
I don't want to change the playlists, just that the users can play them. Or at least list the songs in them.

**Reply by ForMiSoft** [reply]
My android app uses the authorization flow, so I know that it's working, I can get my playlists without a problem.

**Reply by stefanmoro** [reply]
Hey @OlofT ,

You should be able to login users, retrieve their playlists and list the content of those playlists with the functionality exposed today. We also just started work on modify playlists and "my collection".

**Comment by OlofT** [comment]
**SOLVED**
For anyone else having the same problem: the documentation is wrong, loggin in a user works just fine even for the rest of us. But you can't play songs yet.
(In my case I had missed setting up OAuth under "settings" in the dashboard/app. You must use the redirect url you specify there... 🤦🏻‍♂️)

**Reply by stefanmoro** [reply]
Hey @OlofT , can you please point me to the part of the documentation that is incorrect so we can fix that?

Are you referring to documentation at [developer.tidal.com](https://developer.tidal.com/documentation/authorization/authorization-overview), or the [Android SDK auth](https://github.com/tidal-music/tidal-sdk-android/tree/main/auth) README, or somewhere else?

The only place I found that comment you refer to ("Only available for TIDAL internally developed applications for now") is under device login flow in the README. That is correct, that flow is not available for 3rd party apps as of now, but I cannot find any place where the documentation says that logging in users is not available for 3rd party apps.

Thanks

**Reply by OlofT** [reply]
The README, https://github.com/tidal-music/tidal-sdk-ios/blob/main/Sources/Auth/README.md
The "Authorization Code Flow (user login)" section.

**Reply by stefanmoro** [reply]
Thanks, will make sure to fix and align the iOS README with all other docs.
